From 57bdcf78fb74020541a9367ea430cf915b9fed3a Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Mon, 13 Mar 2017 22:49:18 +0100
Subject: [PATCH] media/jni: Check mute state before starting play

When playing in SoundPool, PlayerBase checks
hasAppOpsPlayAudio and calls mute function.
After that, if a new audiotrack is created,
sets the volume regardless of whether mMuted.

diff --git a/media/jni/soundpool/SoundPool.cpp b/media/jni/soundpool/SoundPool.cpp
index 5f1cf160894..05faa3d04f4 100644
--- a/media/jni/soundpool/SoundPool.cpp
+++ b/media/jni/soundpool/SoundPool.cpp
@@ -812,7 +812,11 @@ void SoundChannel::play(const sp<Sample>& sample, int nextChannelID, float leftV
             mAudioTrack = newTrack;
             ALOGV("using new track %p for sample %d", newTrack.get(), sample->sampleID());
         }
-        newTrack->setVolume(leftVolume, rightVolume);
+        if (mMuted) {
+            newTrack->setVolume(0.0f, 0.0f);
+        } else {
+            newTrack->setVolume(leftVolume, rightVolume);
+        }
         newTrack->setLoop(0, frameCount, loop);
         mPos = 0;
         mSample = sample;
